okhttp-tests/src/test/java/com/squareup/okhttp/internal/http/ThreadInterruptTest.java

```diff
@@ -70,6 +92,35 @@ protected void configureSocket(Socket socket) throws IOException {
     });
   }
 
+  @Test @Ignore public void interruptConnect() throws Exception {
```

Comment:
In this case we know that's not happening because of the way the MockWebServer is configured, but you're right that is not tested.

A lot of the complexity of this change comes about from trying to force blocking behavior so we can then timeout or interrupt it. Not very happy with the test but it seemed better than no test and confirmed the code was broken before and better after.

Forego the test instead?